engine/src/main/java/org/terasology/logic/title/TitleSystem.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2018 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.logic.title;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.assets.ResourceUrn;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.logic.console.Console;
+import org.terasology.logic.console.CoreMessageType;
+import org.terasology.logic.console.commandSystem.annotations.Command;
+import org.terasology.logic.console.commandSystem.annotations.CommandParam;
+import org.terasology.logic.delay.DelayManager;
+import org.terasology.logic.delay.DelayedActionTriggeredEvent;
+import org.terasology.logic.players.LocalPlayer;
+import org.terasology.registry.In;
+import org.terasology.registry.Share;
+import org.terasology.rendering.nui.NUIManager;
+
+@Share(Title.class)
+@RegisterSystem(RegisterMode.ALWAYS)
+public class TitleSystem extends BaseComponentSystem implements Title {
+
+    private static final Logger logger = LoggerFactory.getLogger(TitleSystem.class);
+
+    private static final ResourceUrn UI_URL = new ResourceUrn("engine:title");
+
+    private static final String DELAY_ACTION_ID = "TITLE";
+
+    @In
+    private Console console;
+
+    @In
+    private NUIManager nuiManager;
+
+    @In
+    private DelayManager delayManager;
+
+    @In
+    private LocalPlayer localPlayer;
+
+    private TitleScreen titleScreen;
+    private String lastTitle = "";
+    private String lastSubtitle = "";
+    private long lastStay;
+
+    @Override
+    public void initialise() {
+        logger.info("Initialised the title system!");
```

Comment:
Like I mentioned in the AFK PR, ideally these shouldn't be present or at the least debug level.

Reply:
Good note here 😄